src/systems/force_torque/ForceTorque.cc

```diff
@@ -176,15 +180,13 @@ void ForceTorque::PostUpdate(const UpdateInfo &_info,
   if (!_info.paused)
   {
     // check to see if update is necessary
-    // we only update if there is at least one sensor that needs data
-    // and that sensor has subscribers.
+    // we only update if there is at least one sensor that needs data.
     // note: gz-sensors does its own throttling. Here the check is mainly
     // to avoid doing work in the ForceTorquePrivate::Update function
     bool needsUpdate = false;
     for (const auto &[sensorEntity, sensor] : this->dataPtr->entitySensorMap)
     {
-      if (sensor->NextDataUpdateTime() <= _info.simTime &&
-          sensor->HasConnections())
+      if (sensor->NextDataUpdateTime() <= _info.simTime)
```

Comment:
does it make sense for users to indicate that they want data to be written to ECM by creating a `components::WrenchMeasured` component? here we would just check for the presence of that component, e.g. 

```cpp
    if (sensor->NextDataUpdateTime() <= _info.simTime &&
        (sensor->HasConnections() ||
        _ecm.Component<components::WrenchMeasured>(sensorEntity) != nullptr))
```

Probably not going to make much perf difference for this system since the sensor `Update` call should be relatively fast and same for writing force torque values to ECM. It would be nice to keep things consistent if we do decide to port this functionality to other sensors, e.g. contact sensor

Reply:
ooh, I like that. Let me give that a try

Reply:
https://github.com/gazebosim/gz-sim/pull/2494/commits/c6fedb12d30f41e593061d54bd549f94ff447787